src/data/fintype.lean

```diff
@@ -135,6 +135,18 @@ theorem card_of_subtype {p : α → Prop} (s : finset α)
   card {x // p x} = s.card :=
 by rw ← subtype_card s H; congr
 
+/-- Construct a fintype from a finset with the same elements. -/
+def of_finset {p : set α} (s : finset α) (H : ∀ x, x ∈ s ↔ x ∈ p) : fintype p :=
```

Comment:
Should this be called `finset.to_fintype` to allow projection notation?

Reply:
I'm not convinced. We end up producing `fintype p`, for some other `p`, so writing `s.to_fintype H` feels a bit misleading: that should produce a "`s` as a fintype" in some sense.

Another option would be to make `s` an implicit argument as well, but that may harm readability.